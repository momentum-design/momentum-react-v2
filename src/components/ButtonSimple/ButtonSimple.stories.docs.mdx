The `<ButtonSimple />` component. This component is used when injecting buttons into other
components that implement specific styling for their button components. This component should not be
used unless it is being passed in as a prop of another **Momentum UI** component.
