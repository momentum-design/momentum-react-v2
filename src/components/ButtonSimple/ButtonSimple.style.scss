.md-button-simple-wrapper {
  > span,
  .md-text-wrapper {
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }
}
