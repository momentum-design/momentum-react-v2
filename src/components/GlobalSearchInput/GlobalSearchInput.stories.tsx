import { MultiTemplate, Template } from '../../storybook/helper.stories.templates';
import { DocumentationPage } from '../../storybook/helper.stories.docs';
import StyleDocs from '../../storybook/docs.stories.style.mdx';

import GlobalSearchInput, { GlobalSearchInputProps, GlobalSearchInputSearchFilter } from './';
import argTypes from './GlobalSearchInput.stories.args';
import Documentation from './GlobalSearchInput.stories.docs.mdx';
import React, { useState, FC } from 'react';

export default {
  title: 'Momentum UI/GlobalSearchInput',
  component: GlobalSearchInput,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
  },
  args: {
    // Args provided to all stories by default.
    placeholder: 'Search',
    initialText: 'message',
    initialLabel: 'Search messages by person or space',
    clearButtonAriaLabel: 'CLEAR',
    initialFilters: [
      {
        term: 'from',
        value: 'Joe',
        translations: {
          filterRemoved: 'From Joe removed',
          filterAdded: 'From filter added',
          text: 'From: Joe',
          empty: 'Choose who to filter by',
          nonempty: 'Filtering by Joe',
        },
      },
      {
        term: 'in',
        value: 'a space',
        translations: {
          filterRemoved: 'In a space removed',
          filterAdded: 'In filter added',
          text: 'In: a space',
          empty: 'Choose a space to filter by',
          nonempty: 'Filtering in a space',
        },
      },
    ],
    searching: false,
  },
};

interface GlobalSearchInputExampleProps extends GlobalSearchInputProps {
  initialText: string;
  initialFilters: GlobalSearchInputSearchFilter[];
}

const BetterExample: FC<GlobalSearchInputExampleProps> = (props: GlobalSearchInputExampleProps) => {
  const { initialText, initialFilters, initialLabel } = props;

  const mutatedProps = { ...props };
  delete mutatedProps.initialText;
  delete mutatedProps.initialFilters;

  const [val, setVal] = useState(initialText);
  const [filters, setFilters] = useState(initialFilters);
  const [searching, setSearching] = useState(false);

  const handleChange = (e: string) => {
    setSearching(true);
    setTimeout(() => {
      setSearching(false);
    }, 500);
    if (e.toLowerCase().startsWith('from:')) {
      setFilters([
        {
          term: 'from',
          value: '',
          translations: {
            filterAdded: 'From filter added',
            filterRemoved: 'From filter removed',
            text: 'From:',
            empty: 'Choose who to filter by',
            nonempty: '',
          },
        },
        ...filters,
      ]);
      setVal(e.slice('From:'.length, e.length));
    } else {
      setVal(e);
    }
  };

  const handleFiltersChange = (filters) => {
    setFilters(filters);
  };

  return (
    <div style={{ width: '40rem' }}>
      <GlobalSearchInput
        value={val}
        filters={filters}
        onFiltersChange={handleFiltersChange}
        onChange={handleChange}
        initialLabel={initialLabel}
        {...mutatedProps}
        searching={searching}
      />
    </div>
  );
};

/**
 * Primary story. This renders a single component with all external props.
 */
const Example = Template<GlobalSearchInputExampleProps>(BetterExample);

Example.argTypes = { ...argTypes };

/**
 * Common variants story. This renders multiple variants of a single component.
 */

const PaddedExample: FC<GlobalSearchInputProps> = (props: GlobalSearchInputProps) => {
  return (
    <div style={{ margin: '1rem', width: '20rem' }}>
      <GlobalSearchInput {...props} />
    </div>
  );
};

const Common = MultiTemplate<GlobalSearchInputProps>(PaddedExample);

Common.argTypes = { ...argTypes };

Common.parameters = {
  variants: [
    { value: 'Typed' },
    { searching: true, placeholder: 'Searching...' },
    { disabled: true },
    {
      value: 'Joe',
      filters: [
        {
          term: 'from',
          value: '',
          translations: {
            filterAdded: 'From filter added',
            filterRemoved: 'From filter removed',
            text: 'From:',
            empty: 'Choose who to filter by',
            nonempty: '',
          },
        },
      ],
    },
    {
      value: '',
      filters: [
        {
          term: 'from',
          value: 'Joe',
          translations: {
            filterAdded: 'From filter added',
            filterRemoved: 'From filter removed',
            text: 'From: Joe',
            empty: 'Choose who to filter by',
            nonempty: 'Filtering by Joe',
          },
        },
      ],
    },
    {
      value: 'a message',
      filters: [
        {
          term: 'in',
          value: 'Joe',
          translations: {
            filterAdded: 'In filter added',
            filterRemoved: 'In filter removed',
            text: 'In: A Space',
            empty: 'Choose a space to filter by',
            nonempty: 'Filtering in a space',
          },
        },
      ],
    },
    {
      value: 'search term',
      filters: [
        {
          term: 'in',
          value: 'A really long filter name that is longer than the input width',
          translations: {
            filterAdded: 'In filter added',
            filterRemoved: 'In filter removed',
            text: 'In: A really long filter name that is longer than the input width',
            empty: 'Choose a space to filter by',
            nonempty: '',
          },
        },
      ],
    },
  ],
};

export { Example, Common };
