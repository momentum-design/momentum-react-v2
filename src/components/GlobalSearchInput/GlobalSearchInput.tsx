/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { FC, useRef, useState, useEffect } from 'react';
import classnames from 'classnames';

import ButtonSimple from '../ButtonSimple';
import { STYLE } from './GlobalSearchInput.constants';
import { Props } from './GlobalSearchInput.types';
import './GlobalSearchInput.style.scss';
import { useSearchField } from '@react-aria/searchfield';
import { useSearchFieldState } from '@react-stately/searchfield';
import { difference } from 'lodash';

import Icon from '../Icon';
import { BaseEvent } from '@react-types/shared';
import { useFocusState } from '../../hooks/useFocusState';
/**
 * Global search input. Used for global search only
 */
const GlobalSearchInput: FC<Props> = (props: Props) => {
  const {
    initialLabel = '',
    className,
    id,
    style,
    searching,
    onKeyDown,
    onFiltersChange,
    filters = [],
    clearButtonAriaLabel,
    onClear,
  } = props;
  const [previousFilters, setPreviousFilters] = useState(filters);
  const { isFocused, focusProps } = useFocusState(props);

  const [ariaAlert, setAriaAlert] = useState('');
  const state = useSearchFieldState(props);
  const ref = useRef(null);

  useEffect(() => {
    const newFilters = difference(filters, previousFilters);
    if (newFilters.length) {
      setAriaAlert(newFilters[0].translations.filterAdded);
    }
    setPreviousFilters(filters);
  }, [JSON.stringify(filters)]);

  const onClearPress = () => {
    onFiltersChange([]);
    onClear();
  };

  const { inputProps, clearButtonProps, labelProps } = useSearchField(
    { ...props, placeholder: filters.length ? '' : props.placeholder, onClear: onClearPress },
    state,
    ref
  );

  // react aria useSearchField is setting excludeFromTabOrder to true
  // check https://github.com/adobe/react-spectrum/blob/02f242f41f0347b3f11e50b019c197127b7d0a08/packages/%40react-aria/searchfield/src/useSearchField.ts#L118
  // but clear button should be focusable, so we overwrite clearButtonProps.excludeFromTabOrder = false to false
  clearButtonProps.excludeFromTabOrder = false;

  const handleClick = () => {
    if (ref.current) {
      ref.current.focus();
    }
  };

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    const { key } = e;
    const target = e.target as HTMLInputElement;
    if (key === 'Backspace') {
      if (!(target.selectionStart === 0 && target.selectionEnd === 0)) {
        return;
      }
      const filterToRemove = filters[filters.length - 1];
      onFiltersChange && filters.length && onFiltersChange(filters.slice(0, filters.length - 1));
      filterToRemove && setAriaAlert(filterToRemove.translations.filterRemoved);
    } else {
      if (onKeyDown) {
        onKeyDown(e as BaseEvent<React.KeyboardEvent<HTMLInputElement>>);
      }
      inputProps.onKeyDown(e);
    }
  };

  const buildFilters = () => {
    const filterArray = [];
    const labels = [];
    filters.forEach((filter) => {
      labels.push(filter.value ? filter.translations.nonempty : filter.translations.empty);

      filterArray.push(
        <div key={filter.term} className={STYLE.searchContext}>
          <p>{filter.translations.text}</p>
        </div>
      );
    });

    const label = labels.length ? `${labels.join(', ')}:` : initialLabel;

    return { filterArray, label };
  };

  const { label, filterArray } = buildFilters();

  return (
    <div
      className={classnames(className, STYLE.wrapper)}
      id={id}
      onClick={handleClick}
      style={style}
      data-focus={isFocused}
    >
      <label htmlFor={inputProps.id} {...labelProps}>
        {label}
      </label>
      <div>
        <Icon
          weight="light"
          scale={18}
          className={STYLE.search}
          strokeColor="var(--mds-color-theme-text-primary-normal)"
          name={searching ? 'spinner' : 'search'}
        />
      </div>
      <div className={STYLE.container}>
        {filterArray}
        <input {...inputProps} {...focusProps} ref={ref} onKeyDown={handleKeyDown} />
        {ariaAlert && (
          <div className="aria-alert" aria-live="assertive" role="alert">
            {ariaAlert}
          </div>
        )}
      </div>
      {(!!state.value || !!filters.length) && (
        <ButtonSimple
          className={STYLE.clear}
          {...clearButtonProps}
          aria-label={clearButtonAriaLabel}
        >
          <Icon scale={18} name="cancel" />
        </ButtonSimple>
      )}
    </div>
  );
};

export default GlobalSearchInput;
