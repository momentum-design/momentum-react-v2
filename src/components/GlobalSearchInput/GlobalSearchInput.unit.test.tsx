import React, { useState } from 'react';
import { SSRProvider } from '@react-aria/ssr';

import { mountAndWait } from '../../../test/utils';
import Icon from '../Icon';
import GlobalSearchInput, { GLOBAL_SEARCH_INPUT_CONSTANTS as CONSTANTS } from './';
import { act } from 'react-dom/test-utils';
import ButtonSimple from '../ButtonSimple';

const testTranslations = {
  empty: 'empty',
  nonempty: 'nonempty',
  filterAdded: 'filteradded',
  filterRemoved: 'filterremoved',
  text: 'text',
};

describe('<GlobalSearchInput />', () => {
  describe('snapshot', () => {
    const mountComponent = async (component) => {
      const container = await mountAndWait(<SSRProvider>{component}</SSRProvider>);
      return container;
    };

    it('should match snapshot', async () => {
      expect.assertions(1);

      const container = await mountComponent(
        <GlobalSearchInput aria-label="global search" clearButtonAriaLabel="search" />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = await mountComponent(
        <GlobalSearchInput
          aria-label="global search"
          className={className}
          clearButtonAriaLabel="search"
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', async () => {
      expect.assertions(1);

      const id = 'example-id';

      const container = await mountComponent(
        <GlobalSearchInput aria-label="global search" id={id} clearButtonAriaLabel="search" />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      const container = await mountComponent(
        <GlobalSearchInput aria-label="global search" style={style} clearButtonAriaLabel="search" />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when searching', async () => {
      expect.assertions(1);

      const container = await mountComponent(
        <GlobalSearchInput
          aria-label="global search"
          searching={true}
          clearButtonAriaLabel="search"
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when filters are set', async () => {
      expect.assertions(1);

      const container = await mountComponent(
        <GlobalSearchInput
          aria-label="global search"
          value="From: someone"
          filters={[
            {
              term: 'from',
              value: 'Joe',
              translations: testTranslations,
            },
            {
              term: 'in',
              value: 'a space',
              translations: testTranslations,
            },
          ]}
          clearButtonAriaLabel="search"
        />
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', async () => {
      expect.assertions(1);

      const element = (
        await mountAndWait(
          <GlobalSearchInput aria-label="global search" clearButtonAriaLabel="search" />
        )
      )
        .find(GlobalSearchInput)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', async () => {
      expect.assertions(1);

      const className = 'example-class';

      const element = (
        await mountAndWait(
          <GlobalSearchInput
            aria-label="global search"
            className={className}
            clearButtonAriaLabel="search"
          />
        )
      )
        .find(GlobalSearchInput)
        .getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', async () => {
      expect.assertions(1);

      const id = 'example-id-2';

      const element = (
        await mountAndWait(
          <GlobalSearchInput aria-label="global search" id={id} clearButtonAriaLabel="search" />
        )
      )
        .find(GlobalSearchInput)
        .getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', async () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const element = (
        await mountAndWait(
          <GlobalSearchInput
            aria-label="global search"
            style={style}
            clearButtonAriaLabel="clear"
          />
        )
      )
        .find(GlobalSearchInput)
        .getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should pass the aria label to the input', async () => {
      expect.assertions(1);

      const element = (
        await mountAndWait(
          <GlobalSearchInput
            aria-label="global search"
            searching={true}
            clearButtonAriaLabel="clear"
          />
        )
      )
        .find('input')
        .getDOMNode();

      expect(element.getAttribute('aria-label')).toBe('global search');
    });

    it('should pass the aria label to the input', async () => {
      expect.assertions(1);

      const element = (
        await mountAndWait(
          <GlobalSearchInput
            aria-label="global search"
            searching={true}
            clearButtonAriaLabel="clear"
          />
        )
      )
        .find('input')
        .getDOMNode();

      expect(element.getAttribute('aria-label')).toBe('global search');
    });

    it('should pass the correct properties to the clear button', async () => {
      expect.assertions(1);

      const element = (
        await mountAndWait(
          <GlobalSearchInput
            aria-label="global search"
            searching={true}
            clearButtonAriaLabel="clear"
            value="hello"
          />
        )
      ).find(ButtonSimple);

      expect(element.props()).toStrictEqual({
        className: 'md-global-search-input-clear',
        excludeFromTabOrder: false,
        'aria-label': 'clear',
        children: expect.any(Object),
        onPress: expect.any(Function),
        onPressStart: expect.any(Function),
        preventFocusOnPress: true,
      });
    });
  });

  describe('actions', () => {
    it('adding a filter should change the aria alert', async () => {
      expect.assertions(3);

      const onChange = jest.fn();
      const onFiltersChange = jest.fn();

      let setParentFilters;

      const ParentComponent = () => {
        const [filters, setFilters] = useState([]);
        setParentFilters = setFilters;

        return (
          <GlobalSearchInput
            onChange={onChange}
            onFiltersChange={onFiltersChange}
            aria-label="global search"
            value="ab"
            filters={filters}
            clearButtonAriaLabel="clear"
          />
        );
      };

      const wrapper = await mountAndWait(<ParentComponent />);
      const inputElement = wrapper.find('input');

      const domNode = inputElement.getDOMNode() as HTMLInputElement;

      domNode.setSelectionRange(0, 0);

      await act(async () => {
        setParentFilters([{ term: 'from', value: '', translations: testTranslations }]);
      });

      wrapper.update();
      const ariaAlert = wrapper.find('.aria-alert');

      expect(ariaAlert.text()).toEqual('filteradded');
      expect(onChange).not.toHaveBeenCalled();
      expect(onFiltersChange).not.toHaveBeenCalled();
    });

    it('pressing backspace should delete a filter if the cursor is at the beginning', async () => {
      expect.assertions(4);

      const onChange = jest.fn();
      const onFiltersChange = jest.fn();

      const wrapper = await mountAndWait(
        <GlobalSearchInput
          onChange={onChange}
          onFiltersChange={onFiltersChange}
          aria-label="global search"
          value="ab"
          filters={[{ term: 'from', value: '', translations: testTranslations }]}
          clearButtonAriaLabel="clear"
        />
      );
      const inputElement = wrapper.find('input');

      const domNode = inputElement.getDOMNode() as HTMLInputElement;

      domNode.setSelectionRange(0, 0);

      await act(async () => {
        inputElement.simulate('keydown', { key: 'Backspace' });
      });

      wrapper.update();
      const ariaAlert = wrapper.find('.aria-alert');

      expect(ariaAlert.text()).toEqual('filterremoved');
      expect(onChange).not.toHaveBeenCalled();
      expect(onFiltersChange).toBeCalledTimes(1);
      expect(onFiltersChange).toBeCalledWith([]);
    });

    it('pressing backspace should not update alert if no filters', async () => {
      expect.assertions(3);

      const onChange = jest.fn();
      const onFiltersChange = jest.fn();

      const wrapper = await mountAndWait(
        <GlobalSearchInput
          onChange={onChange}
          onFiltersChange={onFiltersChange}
          aria-label="global search"
          value=""
          filters={[]}
          clearButtonAriaLabel="clear"
        />
      );
      const inputElement = wrapper.find('input');

      const domNode = inputElement.getDOMNode() as HTMLInputElement;

      domNode.setSelectionRange(0, 0);

      await act(async () => {
        inputElement.simulate('keydown', { key: 'Backspace' });
      });

      wrapper.update();
      expect(wrapper.find('.aria-alert').exists()).toEqual(false);
      expect(onChange).not.toHaveBeenCalled();
      expect(onFiltersChange).not.toHaveBeenCalled();
    });

    it('pressing backspace should not delete a filter if the cursor is not at the beginning', async () => {
      expect.assertions(2);

      const onChange = jest.fn();
      const onFiltersChange = jest.fn();

      const inputElement = (
        await mountAndWait(
          <GlobalSearchInput
            onChange={onChange}
            onFiltersChange={onFiltersChange}
            aria-label="global search"
            value="ab"
            filters={[{ term: 'from', value: '', translations: testTranslations }]}
            clearButtonAriaLabel="clear"
          />
        )
      ).find('input');

      const domNode = inputElement.getDOMNode() as HTMLInputElement;

      domNode.setSelectionRange(2, 2);

      await act(async () => {
        inputElement.simulate('keydown', { key: 'Backspace' });
      });

      expect(onChange).not.toHaveBeenCalled();
      expect(onFiltersChange).not.toHaveBeenCalled();
    });

    it('pressing backspace should remove a keyword if the cursor is next to it', async () => {
      expect.assertions(3);

      const onChange = jest.fn();
      const onFiltersChange = jest.fn();

      const filters = [
        { term: 'from', value: '', translations: testTranslations },
        { term: 'in', value: '', translations: testTranslations },
      ];

      const inputElement = (
        await mountAndWait(
          <GlobalSearchInput
            onChange={onChange}
            onFiltersChange={onFiltersChange}
            aria-label="global search"
            value="ab cd"
            filters={filters}
            clearButtonAriaLabel="clear"
          />
        )
      ).find('input');

      const domNode = inputElement.getDOMNode() as HTMLInputElement;

      domNode.setSelectionRange(0, 0);

      await act(async () => {
        inputElement.simulate('keydown', { key: 'Backspace' });
      });

      expect(onChange).not.toHaveBeenCalled();
      expect(onFiltersChange).toBeCalledTimes(1);
      expect(onFiltersChange).toBeCalledWith(filters.slice(0, 1));
    });

    it('pressing backspace should not clear input if not deleting the last keyword', async () => {
      expect.assertions(2);

      const onChange = jest.fn();
      const onFiltersChange = jest.fn();

      const inputElement = (
        await mountAndWait(
          <GlobalSearchInput
            onChange={onChange}
            onFiltersChange={onFiltersChange}
            aria-label="global search"
            value="abc"
            filters={[{ term: 'from', value: '', translations: testTranslations }]}
            clearButtonAriaLabel="clear"
          />
        )
      ).find('input');

      const domNode = inputElement.getDOMNode() as HTMLInputElement;

      domNode.setSelectionRange(2, 2);

      await act(async () => {
        inputElement.simulate('keydown', { key: 'Backspace' });
      });

      expect(onChange).not.toHaveBeenCalled();
      expect(onFiltersChange).not.toHaveBeenCalled();
    });

    it('clicking on another part of the component gives focus to the input', async () => {
      expect.assertions(1);

      const wrapper = await mountAndWait(
        <GlobalSearchInput aria-label="global search" clearButtonAriaLabel="clear" />
      );

      const inputElement = wrapper.find('input');
      const icon = wrapper.find(Icon);

      const domNode = inputElement.getDOMNode() as HTMLInputElement;
      const focusSpy = jest.spyOn(domNode, 'focus');

      domNode.setSelectionRange(2, 2);

      await act(async () => {
        icon.simulate('click');
      });

      expect(focusSpy).toBeCalledWith();
    });
  });
});
