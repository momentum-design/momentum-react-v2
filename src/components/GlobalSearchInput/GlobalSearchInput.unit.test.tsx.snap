// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalSearchInput /> snapshot should match snapshot 1`] = `
<SSRProvider>
  <GlobalSearchInput
    aria-label="global search"
    clearButtonAriaLabel="search"
  >
    <div
      className="md-global-search-input-wrapper"
      data-focus={false}
      onClick={[Function]}
    >
      <label
        htmlFor="test-ID"
      />
      <div>
        <Icon
          className="md-global-search-input-search"
          name="search"
          scale={18}
          strokeColor="var(--mds-color-theme-text-primary-normal)"
          weight="light"
        >
          <div
            aria-hidden="true"
            className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-global-search-input-search md-icon-no-shrink"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-autoscale={false}
              data-scale={18}
              data-test="search"
              fill="currentColor"
              height="100%"
              style={
                {
                  "stroke": "var(--mds-color-theme-text-primary-normal)",
                }
              }
              viewBox="0, 0, 32, 32"
              width="100%"
            />
          </div>
        </Icon>
      </div>
      <div
        className="md-global-search-input-container"
      >
        <input
          aria-label="global search"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="search"
          value=""
        />
      </div>
    </div>
  </GlobalSearchInput>
</SSRProvider>
`;

exports[`<GlobalSearchInput /> snapshot should match snapshot when filters are set 1`] = `
<SSRProvider>
  <GlobalSearchInput
    aria-label="global search"
    clearButtonAriaLabel="search"
    filters={
      [
        {
          "term": "from",
          "translations": {
            "empty": "empty",
            "filterAdded": "filteradded",
            "filterRemoved": "filterremoved",
            "nonempty": "nonempty",
            "text": "text",
          },
          "value": "Joe",
        },
        {
          "term": "in",
          "translations": {
            "empty": "empty",
            "filterAdded": "filteradded",
            "filterRemoved": "filterremoved",
            "nonempty": "nonempty",
            "text": "text",
          },
          "value": "a space",
        },
      ]
    }
    value="From: someone"
  >
    <div
      className="md-global-search-input-wrapper"
      data-focus={false}
      onClick={[Function]}
    >
      <label
        htmlFor="test-ID"
      >
        nonempty, nonempty:
      </label>
      <div>
        <Icon
          className="md-global-search-input-search"
          name="search"
          scale={18}
          strokeColor="var(--mds-color-theme-text-primary-normal)"
          weight="light"
        >
          <div
            aria-hidden="true"
            className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-global-search-input-search md-icon-no-shrink"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-autoscale={false}
              data-scale={18}
              data-test="search"
              fill="currentColor"
              height="100%"
              style={
                {
                  "stroke": "var(--mds-color-theme-text-primary-normal)",
                }
              }
              viewBox="0, 0, 32, 32"
              width="100%"
            />
          </div>
        </Icon>
      </div>
      <div
        className="md-global-search-input-container"
      >
        <div
          className="md-global-search-input-search-context"
          key="from"
        >
          <p>
            text
          </p>
        </div>
        <div
          className="md-global-search-input-search-context"
          key="in"
        >
          <p>
            text
          </p>
        </div>
        <input
          aria-label="global search"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          type="search"
          value="From: someone"
        />
      </div>
      <ButtonSimple
        aria-label="search"
        className="md-global-search-input-clear"
        excludeFromTabOrder={false}
        onPress={[Function]}
        onPressStart={[Function]}
        preventFocusOnPress={true}
      >
        <FocusRing>
          <FocusRing
            focusClass="md-focus-ring-wrapper children"
            focusRingClass="md-focus-ring-wrapper children"
          >
            <button
              aria-label="search"
              className="md-global-search-input-clear md-button-simple-wrapper"
              onBlur={[Function]}
              onClick={[Function]}
              onDragStart={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <Icon
                name="cancel"
                scale={18}
              >
                <div
                  aria-hidden="true"
                  className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-autoscale={false}
                    data-scale={18}
                    data-test="cancel"
                    fill="currentColor"
                    height="100%"
                    style={{}}
                    viewBox="0, 0, 32, 32"
                    width="100%"
                  />
                </div>
              </Icon>
            </button>
          </FocusRing>
        </FocusRing>
      </ButtonSimple>
    </div>
  </GlobalSearchInput>
</SSRProvider>
`;

exports[`<GlobalSearchInput /> snapshot should match snapshot when searching 1`] = `
<SSRProvider>
  <GlobalSearchInput
    aria-label="global search"
    clearButtonAriaLabel="search"
    searching={true}
  >
    <div
      className="md-global-search-input-wrapper"
      data-focus={false}
      onClick={[Function]}
    >
      <label
        htmlFor="test-ID"
      />
      <div>
        <Icon
          className="md-global-search-input-search"
          name="spinner"
          scale={18}
          strokeColor="var(--mds-color-theme-text-primary-normal)"
          weight="light"
        >
          <div
            aria-hidden="true"
            className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-global-search-input-search md-icon-no-shrink"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-autoscale={false}
              data-scale={18}
              data-test="spinner"
              fill="currentColor"
              height="100%"
              style={
                {
                  "stroke": "var(--mds-color-theme-text-primary-normal)",
                }
              }
              viewBox="0, 0, 32, 32"
              width="100%"
            />
          </div>
        </Icon>
      </div>
      <div
        className="md-global-search-input-container"
      >
        <input
          aria-label="global search"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="search"
          value=""
        />
      </div>
    </div>
  </GlobalSearchInput>
</SSRProvider>
`;

exports[`<GlobalSearchInput /> snapshot should match snapshot with className 1`] = `
<SSRProvider>
  <GlobalSearchInput
    aria-label="global search"
    className="example-class"
    clearButtonAriaLabel="search"
  >
    <div
      className="example-class md-global-search-input-wrapper"
      data-focus={false}
      onClick={[Function]}
    >
      <label
        htmlFor="test-ID"
      />
      <div>
        <Icon
          className="md-global-search-input-search"
          name="search"
          scale={18}
          strokeColor="var(--mds-color-theme-text-primary-normal)"
          weight="light"
        >
          <div
            aria-hidden="true"
            className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-global-search-input-search md-icon-no-shrink"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-autoscale={false}
              data-scale={18}
              data-test="search"
              fill="currentColor"
              height="100%"
              style={
                {
                  "stroke": "var(--mds-color-theme-text-primary-normal)",
                }
              }
              viewBox="0, 0, 32, 32"
              width="100%"
            />
          </div>
        </Icon>
      </div>
      <div
        className="md-global-search-input-container"
      >
        <input
          aria-label="global search"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="search"
          value=""
        />
      </div>
    </div>
  </GlobalSearchInput>
</SSRProvider>
`;

exports[`<GlobalSearchInput /> snapshot should match snapshot with id 1`] = `
<SSRProvider>
  <GlobalSearchInput
    aria-label="global search"
    clearButtonAriaLabel="search"
    id="example-id"
  >
    <div
      className="md-global-search-input-wrapper"
      data-focus={false}
      id="example-id"
      onClick={[Function]}
    >
      <label
        htmlFor="example-id"
      />
      <div>
        <Icon
          className="md-global-search-input-search"
          name="search"
          scale={18}
          strokeColor="var(--mds-color-theme-text-primary-normal)"
          weight="light"
        >
          <div
            aria-hidden="true"
            className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-global-search-input-search md-icon-no-shrink"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-autoscale={false}
              data-scale={18}
              data-test="search"
              fill="currentColor"
              height="100%"
              style={
                {
                  "stroke": "var(--mds-color-theme-text-primary-normal)",
                }
              }
              viewBox="0, 0, 32, 32"
              width="100%"
            />
          </div>
        </Icon>
      </div>
      <div
        className="md-global-search-input-container"
      >
        <input
          aria-label="global search"
          disabled={false}
          id="example-id"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="search"
          value=""
        />
      </div>
    </div>
  </GlobalSearchInput>
</SSRProvider>
`;

exports[`<GlobalSearchInput /> snapshot should match snapshot with style 1`] = `
<SSRProvider>
  <GlobalSearchInput
    aria-label="global search"
    clearButtonAriaLabel="search"
    style={
      {
        "color": "pink",
      }
    }
  >
    <div
      className="md-global-search-input-wrapper"
      data-focus={false}
      onClick={[Function]}
      style={
        {
          "color": "pink",
        }
      }
    >
      <label
        htmlFor="test-ID"
      />
      <div>
        <Icon
          className="md-global-search-input-search"
          name="search"
          scale={18}
          strokeColor="var(--mds-color-theme-text-primary-normal)"
          weight="light"
        >
          <div
            aria-hidden="true"
            className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-global-search-input-search md-icon-no-shrink"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-autoscale={false}
              data-scale={18}
              data-test="search"
              fill="currentColor"
              height="100%"
              style={
                {
                  "stroke": "var(--mds-color-theme-text-primary-normal)",
                }
              }
              viewBox="0, 0, 32, 32"
              width="100%"
            />
          </div>
        </Icon>
      </div>
      <div
        className="md-global-search-input-container"
      >
        <input
          aria-label="global search"
          disabled={false}
          id="test-ID"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="search"
          value=""
        />
      </div>
    </div>
  </GlobalSearchInput>
</SSRProvider>
`;
